### PR TITLE
Separate Cloud API from Java/Maven: move non-UI Maven classes and services into their own sub-module

### PR DESCRIPTION
**Pull request by ivanporty**
Progress towards #2182 and #2193, also see detailed plan in #2199.

Moves non-UI, Maven-related independent classes and services into `maven-apis` sub-module.
Since many parts in the UI and tests in the core `google-cloud-apis` still use them, the core API module will for the time being depend on new Maven sub-module. Once all dependencies are cleanly cut using extension points and services, this dependency on maven sub-module will be removed.

### CODE REVIEW

**Comment by patflynn** [comment]
have you all thought about the trade-offs between the accuracy of breaking up modules to this level of granularity vs keeping the modules coarse?  I'm a bit worried that getting super fine grained is going to be  quite cumbersome.

**Comment by etanshaul** [comment]
if the modules are coarse, then we lose the ability to expose features to more IDEs. In other words, if we don't break the cloud-api module down to java and non-java modules, then we cannot expose this to PyCharm for example. or are you suggesting to reconsider whether that's worth doing given the tradeoff?

**Comment by ivanporty** [comment]
In this particular case this immediately gives us significant positive results:
* allows Cloud APIs management to be used without Maven based project
* at the end of refactoring (which shouldn't be quite complicated), we will expose Cloud APIs/service accounts for all other IDEs.
* using extension points, we can add more run configurations for other languages like Go and Python.

I think in this case this looks justified?

**Comment by etanshaul** [comment]
one trade-off we _could_ make here however is to not break apart cloud-api-java and cloud-api-maven into separate modules, and just require that the maven plugin be configured. I'm not saying we should, just that this is a place where we could trim down on the modules

**Comment by etanshaul** [comment]
we would native projects them only if they don't have the maven plugin installed right? whether the project itself is maven/gradle is independent of whether the Maven plugin is installed in there IDE

**Comment by patflynn** [comment]
I didn't think that was true, I believe you can decouple the idea of a IntelliJ module from a plugin fragment.  e.g. one module can have all app engine feature, while optional depends plugin fragments only pull in the features from the module that match the dependency.   The downside to this is that the development classpath for the module won't 100% match what's actually available to the code at runtime which is obviously bad.  

I thought I'd noticed JB seems to follow this convention in the CE examples I've looked at, and looking at the explosion of modules that could come out of creating a submodule for every optional depends I can imagine why. 

I'm actually not really opinionated about this, but I want to make sure we make a conscious decision understanding the trade-offs.

**Comment by patflynn** [comment]
Just to emphasize: I'm very supportive of decoupling the API and secret management from Maven based projects.  I'm just wondering how we should organize our modules to keep us productive.

**Comment by etanshaul** [comment]
@patflynn I see what you are saying now.Mind elaborating on your concern for having fine grained modules?

With the approach you mentioned of keeping optionally included classes in the same module we run into a hazard where a developer can access code that won't be available at runtime, leading to errors that we potentially may not catch right away. That seems like a worse scenario to me.

File: google-cloud-apis/java-apis/maven-apis/resources/META-INF/cloud-apis-java-maven.xml [hunk 1]
```diff
@@ -15,4 +15,33 @@
   -->
 
 <idea-plugin>
+    <depends>org.jetbrains.idea.maven</depends>
+
+    <project-components>
+        <component>
+            <implementation-class>com.google.cloud.tools.intellij.cloudapis.maven.CloudLibraryProjectState
```
Comment (etanshaul) [inline google-cloud-apis/java-apis/maven-apis/resources/META-INF/cloud-apis-java-maven.xml]: Should we rename this to `CloudLibraryMavenProjectState`?

Comment (ivanporty) [reply google-cloud-apis/java-apis/maven-apis/resources/META-INF/cloud-apis-java-maven.xml]: sounds good.

**Pull request merged**